web/packages/teleport/src/Clusters/ManageCluster/ManageCluster.test.tsx

```diff
@@ -0,0 +1,87 @@
+/**
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React from 'react';
+import { MemoryRouter, Route } from 'react-router-dom';
+
+import { render, waitFor, screen } from 'design/utils/testing';
+
+import { ContextProvider } from 'teleport/index';
+import { createTeleportContext } from 'teleport/mocks/contexts';
+import { ContentMinWidth } from 'teleport/Main/Main';
+
+import { clusterInfoFixture } from '../fixtures';
+
+import { ManageCluster } from './ManageCluster';
+
+function renderElement(element, ctx) {
+  return render(
+    <MemoryRouter initialEntries={[`/clusters/cluster-id`]}>
+      <Route path="/clusters/:clusterId">
+        <ContentMinWidth>
+          <ContextProvider ctx={ctx}>{element}</ContextProvider>
+        </ContentMinWidth>
+      </Route>
+    </MemoryRouter>
+  );
+}
+
+describe('test ManageCluster component', () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
+  test('fetches cluster information on load', async () => {
+    const ctx = createTeleportContext();
+    jest
+      .spyOn(ctx.clusterService, 'fetchClusterDetails')
+      .mockResolvedValueOnce({ ...clusterInfoFixture });
```

Comment:
It would be better to mock the API response here with msw instead of spying on internal methods and asserting they were called. Tests more of the underlying code

Reply:
Thanks, updated in https://github.com/gravitational/teleport/pull/47899/commits/4dbdbb3663f8a54b327a1942f20e585cbfef5134